jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -64,23 +71,18 @@
   /** Used for logging during main class inference. */
   private static final String JAR_PLUGIN_NAME = "'jar' task";
 
+  /** Name of the `main` {@link SourceSet} to use as source files. */
+  private static final String MAIN_SOURCE_SET_NAME = "main";
+
   /** @return a GradleProjectProperties from the given project and logger. */
   static GradleProjectProperties getForProject(
       Project project,
       Logger logger,
       Path extraDirectory,
       Map<String, String> permissions,
       AbsoluteUnixPath appRoot) {
-    try {
-      return new GradleProjectProperties(
-          project,
-          logger,
-          GradleLayerConfigurations.getForProject(
-              project, logger, extraDirectory, convertPermissionsMap(permissions), appRoot));
-
-    } catch (IOException ex) {
-      throw new GradleException("Obtaining project build output files failed", ex);
-    }
+    return new GradleProjectProperties(
+        project, logger, extraDirectory, convertPermissionsMap(permissions), appRoot);
```

Comment:
I have an idea. What if we don't pass `extraDirectory` and `permissions` but add the layer outside `ProjectProperties`? I think `extraDirectory` can be not project-specific and independent of Gradle/Mavan. I've not been happy that we have to pass these when instantiating `ProjectProperties`. Also, it's that you add the layer outside `JavaContainerBuilder`, so if it doesn't have a lot of code duplication, it is natual to add the layer separately later?

Reply:
It looks like resolving these is still a bit plugin specific; the project directory is needed to figure out the extra directory (which is different between maven and gradle), and each of the plugins parses the permissions map differently. So it seems like we'll still need to pass these parameters around a bit. But maybe it'd make more sense to do it using `RawConfiguration` instead of `ProjectProperties`, and still handle adding the extra directory in plugins-common.